jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/statistical/MaxFilter.java

```diff
@@ -21,21 +16,9 @@ public MaxFilter(int maskSize) {
         this.maskSize = maskSize;
     }
 
-    private int maskSize;
-
     @Override
     public void apply(int x, int y) {
-        List<Pixel> pixelList = new ArrayList<>();
-
-        for (int i = -maskSize / 2; i <= maskSize / 2; ++i) {
-            for (int j = -maskSize / 2; j <= maskSize / 2; ++j) {
-                try {
-                    pixelList.add(getPixel(x + i, y + j));
-                } catch (Exception e) {
-                    //corner case, no need to handle
-                }
-            }
-        }
+        List<Pixel> pixelList = getPixelList(x, y);
 
         Integer red = pixelList.stream().map(Pixel::getRed).sorted().max(Comparator.naturalOrder()).get();
```

Comment:
and getPixelResult would be implemented in each of statistical filters in a different way to return different pixel